code/modules/mob/living/living.dm

```diff
@@ -160,6 +160,8 @@ Sorry Giacom. Please don't be mad :(
 	set name = "Pull"
 	set category = "Object"
 
+	if(pulling == AM)
+		stop_pulling()

```

Comment:
now you have the opposite issue, if you click the object  you're currently pulling then you will stop pulling it and immediately start pulling it again.
I think you should change `if(AM.Adjacent(src))` to `else if(AM.Adjacent(src))`.